irohad/consensus/yac/impl/yac.cpp

```diff
@@ -139,13 +143,48 @@ namespace iroha {
       void Yac::onState(std::vector<VoteMessage> state) {
         std::unique_lock<std::mutex> guard(mutex_);
 
-        removeUnknownPeersVotes(state);
+        removeUnknownPeersVotes(state, getCurrentOrder());
         if (state.empty()) {
           log_->debug("No votes left in the message.");
           return;
         }
 
         if (crypto_->verify(state)) {
+          auto &proposal_round = getRound(state);
+
+          if (proposal_round.block_round > round_.block_round) {
+            guard.unlock();
+            auto same_hashes =
+                std::all_of(std::next(state.begin()),
+                            state.end(),
+                            [first = state.begin()](const auto &current) {
+                              return first->hash == current.hash;
+                            });
+            auto answer = same_hashes ? Answer(CommitMessage(state))
```

Comment:
I am against calling a message from the future either "commit" or "reject". Can we make some kind of "future message" that will enclose both cases?